### DISCUSSION TITLE
Culture Specific custom date formatting.

### DISCUSSION THREAD

**Discussion by sseyalioglu** [JoshClose/CsvHelper#1561] [Q&A]
Currently I have DTO as the following, which worked pretty fine until now.

        [Index(5), Format("MM-dd-yy")]
        public DateTime BeginDate { get; set; }

        [Index(6), Format("MM-dd-yy")]
        public DateTime EndDate { get; set; }

But now, I started to receive another data that does not fit this. They send dd/MM/yy for en-CA. So by default, that is really no the formatting and so far I used the following:
new CsvConfiguration(CultureInfo.InvariantCulture)

How can I make reader work based on the custom format being sent.
I am willing to make a typeconverter to handle it but seems like I cannot access the culture set in the configuration. 

Can you please guide?

**Comment by JoshClose** [comment]
Instead of `CultureInfo.InvariantCulture` you'll want to use `new CultureInfo("en-CA")`.

**Comment by sseyalioglu** [comment]
But what about the custom format? Currently hard coded as Format("MM-dd-yy") because that is how it is received so far and still receiving that.
For en-CA like you suggested is perfectly fine to put, just want to know how 2 different formats can live together based on culture.

**Comment by sseyalioglu** [comment]
It is possible that I found the solution. Here is the custom converter I was able to write. 

    public class DateTimeConverterCustomCultureDependent : DateTimeConverter
    {
        public override object ConvertFromString(string text, IReaderRow row, MemberMapData memberMapData)
        {
            if (string.IsNullOrEmpty(text))
                return null;

            try
            {
                var convertFormats = new[] {"MM-dd-yy"};
                switch (row.Configuration.CultureInfo.Name)
                {
                    case "en-CA":
                        convertFormats = new[] {"dd/MM/yy"};
                        break;
                }

                foreach (var convertFormat in convertFormats)
                {
                    if (!DateTime.TryParseExact(text, convertFormat, row.Configuration.CultureInfo.DateTimeFormat, DateTimeStyles.None, out var parsedDate))
                        continue;

                    return parsedDate;
                }

                return base.ConvertFromString(text, row, memberMapData);
            }
            catch
            {
                return null;
            }
        }
    }

**Comment by JoshClose** [comment]
You can specify multiple formats.

```cs
[Format("MM-dd-yy", "dd/MM/yy")]
```

**Comment by sseyalioglu** [comment]
Right, I know I can but in that case I cannot understand if 05/08/20 is August or May, that is why I needed culture.
